File: Infinite Scroll (ObjC)/InfiniteScrollSample/SponsoredLandingPageViewController.m

#import "SponsoredLandingPageViewController.h"


@interface SponsoredLandingPageViewController ()
@property (nonatomic) UIBarButtonItem *shareButton;
@property (nonatomic) NtvAdData *adData;
@end

@implementation SponsoredLandingPageViewController


- (void)viewDidLoad
{
    [super viewDidLoad];
    
    //Create the share button
    self.shareButton = [[UIBarButtonItem alloc]
                        initWithBarButtonSystemItem:UIBarButtonSystemItemAction
                        target:self
                        action:@selector(shareButtonTouched)];
    self.navigationItem.rightBarButtonItem = self.shareButton;
}


-(void)shareButtonTouched {
    NSString *socialUrl = [NtvSharing getShareLinkForPlatform:NtvSharePlatformOther withAd:self.adData];
    
    UIActivityViewController *avc = [[UIActivityViewController alloc] initWithActivityItems:@[socialUrl] applicationActivities:nil];
    avc.completionWithItemsHandler = ^(UIActivityType __nullable activityType, BOOL completed, NSArray * __nullable returnedItems, NSError * __nullable activityError) {
        
        NtvSharePlatform sharePlatform = NtvSharePlatformOther;
        if ([activityType isEqualToString:UIActivityTypePostToFacebook] ) {
            sharePlatform = NtvSharePlatformFacebook;
        } else if ([activityType isEqualToString: UIActivityTypePostToTwitter]) {
            sharePlatform = NtvSharePlatformTwitter;
        }
        
        [NtvSharing trackShareActionForPlatform:sharePlatform withAd:self.adData];
    };
    UIViewController *rootVC = [[UIApplication sharedApplication].delegate window].rootViewController;
    [rootVC presentViewController:avc animated:YES completion:nil];
}


#pragma mark - NtvLandingPageInterface

- (void)didLoadContentWithAd:(NtvAdData *)adData {
    self.adData = adData;
}

- (NtvContentWebView *)contentWKWebView {
    return self.nativoWebView;
}

- (NSString *)formatDate:(NSDate *)date {
    NSDateFormatter* dateFormatter = [[NSDateFormatter alloc] init];
    [dateFormatter setDateFormat:@"MM-dd-yyyy"];
    return [dateFormatter stringFromDate:date];
}

- (void)handleExternalLink:(NSURL *)link {
    // Load click-out url in WKWebView
    UIViewController *clickoutAdVC = [[UIViewController alloc] init];
    WKWebView *webView = [[WKWebView alloc] initWithFrame:clickoutAdVC.view.frame];
    [clickoutAdVC.view addSubview:webView];
    NSURLRequest *clickoutReq = [NSURLRequest requestWithURL:link];
    [webView loadRequest: clickoutReq];
    [self.navigationController pushViewController:clickoutAdVC animated:YES];
}


@end

File: GAM Integration (Swift)/NativoGAMIntegrationSample/SponsoredLandingPageViewController.swift
//
//  SponsoredLandingPageViewController.swift
//  NativoGAMIntegrationSample
//

import UIKit
import NativoSDK

class SponsoredLandingPageViewController: UIViewController, NtvLandingPageInterface {
    
    @IBOutlet weak var titleLabel: UILabel!
    @IBOutlet weak var authorNameLabel: UILabel!
    @IBOutlet weak var authorImageView: UIImageView!
    @IBOutlet weak var dateLabel: UILabel!
    @IBOutlet weak var previewTextLabel: UILabel!
    @IBOutlet weak var previewImageView: UIImageView!
    @IBOutlet weak var contentWKWebView: WKWebView!

    var shareUrl: String?
    var trackDidShare: TrackDidShareBlock!
    var shareBtn : UIBarButtonItem?
    var adData : NtvAdData?
    
    func handleExternalLink(_ link: URL) {
        // Load click-out url in WKWebView
        let clickoutAdViewController = UIViewController.init()
        let webView = WKWebView.init(frame: clickoutAdViewController.view.frame)
        clickoutAdViewController.view.addSubview(webView)
        let clickoutReq = URLRequest.init(url: link)
        webView.load(clickoutReq)
        self.navigationController?.pushViewController(clickoutAdViewController, animated: true)
    }
    
    func didLoadContent(withAd adData: NtvAdData) {
        self.adData = adData
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        self.shareBtn = UIBarButtonItem.init(barButtonSystemItem: .action, target: self, action:#selector(socialShareButtonClick))
        self.navigationItem.rightBarButtonItem = self.shareBtn
    }
    
    @objc
    func socialShareButtonClick() {
        if let socialURL = self.shareUrl {
            let avc = UIActivityViewController(activityItems: [socialURL], applicationActivities: nil)
            avc.completionWithItemsHandler = {(activityType: UIActivity.ActivityType?, completed: Bool, returnedItems: [Any]?, activityError: Error?) in
                var sharePlatform = NtvSharePlatform.other
                if activityType == UIActivity.ActivityType.postToFacebook {
                    sharePlatform = NtvSharePlatform.facebook
                }
                else if activityType == UIActivity.ActivityType.postToTwitter {
                    sharePlatform = NtvSharePlatform.twitter
                }
                self.trackDidShare(sharePlatform)
            }
            self.present(avc, animated: true, completion: nil)
        }
    }

}
